BHoMAnalytics_UI/CaptureProjectData.xaml.cs

```diff
@@ -43,58 +43,148 @@ namespace BH.UI.Analytics
+            NonProjectBtn.Focus();
+            NonProjectSelectionPanel.Visibility = Visibility.Visible;
+            this.Height = 380;
+            NonProjectListBox.SelectedIndex = 0;
+            NonProjectListBox.Focus();
         }
 
-        private void Click_ConfirmProjectBtn(object sender, EventArgs e)
+        private void Click_ConfirmNonProjectBtn(object sender, RoutedEventArgs e)
         {
             ConfirmProject();
         }
 
-        private void KeyDown_ProjectIDEntry(object sender, KeyEventArgs e)
+        private void KeyDown_NonProjectListBox(object sender, KeyEventArgs e)
         {
             if (e.Key == Key.Enter || e.Key == Key.Return)
                 ConfirmProject();
         }
 
+        private void NonProjectListBox_SelectionChanged(object sender, SelectionChangedEventArgs e)
+        {
+            if (NonProjectListBox.SelectedValue.Equals(NonProjectOption.Other))
+            {
+                this.Height = 445;
+                OtherSpecifyInput.Visibility = Visibility.Visible;
+                OtherSpecifyText.Visibility = Visibility.Visible;
+            }
+            else
+            {
+                this.Height = 380;
+                OtherSpecifyInput.Visibility = Visibility.Collapsed;
+                OtherSpecifyText.Visibility = Visibility.Collapsed;
+            }
+        }
+        #endregion
+
+        #region Window
+        
         private void Deactivate_Window(object sender, EventArgs e)
         {
             this.Topmost = true;
         }
+        
+        private void Window_MouseDown(object sender, MouseButtonEventArgs e)
+        {
+            if (e.ChangedButton == MouseButton.Left)
+                this.DragMove();
+        }
+        #endregion
+        
+        #endregion
 
+        #region PrivateMethods
+        
         private void ConfirmProject()
         {
-            var projectID = ProjectIDInput.Text;
+            string projectID = "";
+            if (ProjectInputPanel.Visibility == Visibility.Visible)
+            {
+                projectID = ProjectIDInput.Text;
+            }
+            else if (NonProjectSelectionPanel.Visibility == Visibility.Visible && OtherSpecifyInput.Visibility == Visibility.Visible)
+            {
+                projectID = $"Non-Project Work - {(int)NonProjectListBox.SelectedValue} - {OtherSpecifyInput.Text}";
+            }
+            else if (NonProjectSelectionPanel.Visibility == Visibility.Visible)
+            {
+                projectID = $"Non-Project Work - {(int)NonProjectListBox.SelectedValue}";
+            }
+
             if (string.IsNullOrEmpty(projectID))
             {
                 MessageBox.Show("Project ID cannot be empty", "Error", MessageBoxButton.OK);
                 return;
             }
 
+            if ((NonProjectSelectionPanel.Visibility == Visibility.Visible && OtherSpecifyInput.Visibility == Visibility.Visible) && string.IsNullOrEmpty(OtherSpecifyInput.Text))
```

Comment:
I would move this check to be above the building of the `projectID` variable above